src/services/mongoose/users.ts

```diff
@@ -29,6 +27,29 @@ export const UsersRepository = (): IUsersRepository => {
     }
   }
 
+  const findByIdAndUpdateLastConnectionDate = async (
+    userId: UserId,
+    lastConnection: Date,
+  ): Promise<User | RepositoryError> => {
+    try {
+      const result = await User.findOneAndUpdate(
+        {
+          _id: userId,
+        },
+        {
+          lastConnection,
+        },
+      )
+      if (!result) {
+        return new CouldNotFindUserFromIdError(userId)
```

Comment:
should be something like 
```
      if (result.n === 0) {
        return new CouldNotFindUserError()
      }

      if (result.nModified !== 1) {
        return new CouldNotUpdateLastConnectionError()
      }
```

Reply:
it's not an `update` query, it's a `findBy`.